packages/salesforce-adapter/test/filters/currency_iso_code.test.ts

```diff
@@ -0,0 +1,176 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { Element, ObjectType, BuiltinTypes, Field, ElemID } from '@salto-io/adapter-api'
+import { findElements as findElementsByID } from '@salto-io/adapter-utils'
+import currencyIsoCodeFilter from '../../src/filters/currency_iso_code'
+import { FilterWith } from '../../src/filter'
+import { FIELD_TYPE_NAMES, SALESFORCE, CURRENCY_CODE_TYPE_NAME } from '../../src/constants'
+import { Types } from '../../src/transformers/transformer'
+import { createValueSetEntry, createCustomObjectType, findElements } from '../utils'
+
+type PicklistDefinition = {
+  name: string
+  values: string[]
+}
+
+const createMockElemenet = (...picklists: PicklistDefinition[]): ObjectType => {
+  const elem = createCustomObjectType('MockType', {
+    fields: {
+      Id: { refType: BuiltinTypes.SERVICE_ID },
+      Name: { refType: Types.primitiveDataTypes[FIELD_TYPE_NAMES.TEXT] },
+    },
+  })
+
+  picklists.forEach(({ name, values }) => {
+    elem.fields[name] = new Field(
+      elem,
+      name,
+      Types.primitiveDataTypes[FIELD_TYPE_NAMES.PICKLIST],
+      { valueSet: values.map(value => createValueSetEntry(value, false, value, true)) },
+    )
```

Comment:
you can use the constructor, not need to modify the object after creation
(don't that the suggestion as-is, add more line breaks)
```suggestion
  const elem = createCustomObjectType('MockType', {
    fields: {
      Id: { refType: BuiltinTypes.SERVICE_ID },
      Name: { refType: Types.primitiveDataTypes[FIELD_TYPE_NAMES.TEXT] },
      ...picklists.map(({ name, values }) => [name, { refType: Types.primitiveDataTypes[FIELD_TYPE_NAMES.PICKLIST], annotations: { valueSet: values.map(value => createValueSetEntry(value, false, value, true)) } }])
    },
  })
```